### DISCUSSION TITLE
Connection issues with beta 6

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#763] [Help]
Hi,

So I still got 3  kind of issues : 
- unable to connect to localhost server on mac (was working perfectly up to beta 5 yesterday)
- server crash on mac when loading a plugin
- Some AG insert in DAW cant connect to a remote AG server (tried on several win 10 server with beta6).

I already send some information for the first  two cases.
For this second case, I noticed two kinds of error in the logs :

- 17:02:40.183|winpos|23112806e70] **CreateFileA failed:** The process cannot access the file because it is being used by another process.
(I have seen this one for a while in several beta)

- [17:02:42.639|worker|2311496eea0|client:7fc833116dd0] adding plugin VST3-TAUPE-c45c11a...[17:02:42.644|agprocessor|**7ff650efb8b8] failed to find plugin descriptor**
- (plugin is there and working perfectly in cubase on the remote server)

Here are the logs attached

[AudioGridderServer_2021-10-13_12-16-35.log](https://github.com/apohl79/audiogridder/files/7339394/AudioGridderServer_2021-10-13_12-16-35.log)
[Sandbox disconnection.txt](https://github.com/apohl79/audiogridder/files/7339397/Sandbox.disconnection.txt)

**Comment by Henri75** [comment]
Clearly after looking a lot of logs, it seems that in a lot of case AG cant reconnect to server because of this error : 
winpos|23112806e70] CreateFileA failed

Sometimes it works, sometime not. After a lot of disconnection/reconnection ... it may finally reconnect and work :(

**Comment by Henri75** [comment]
And |agprocessor|7ff650efb8b8] failed to find plugin descriptor
both errors appear together when it fails

**Comment by Henri75** [comment]
Plugins are even there in the AG plugin window when I checked

**Comment by apohl79** [comment]
> [17:02:42.639|worker|2311496eea0|client:7fc833116dd0] adding plugin VST3-TAUPE-c45c11a...
> [17:02:42.644|agprocessor|7ff650efb8b8] failed to find plugin descriptor

Did you upgrade that plugin? This is a bit concerning, because beta6 upgrades to the latest JUCE framework which comes with some plugin ID related changes. I know, it is probably a bit of work, but could you do a full plugin scan with beta5 and one with beta6 and provide me with the plugin cache files of AG?

**Comment by Henri75** [comment]
it is relly werid.
When i add a new AG insert and load the plugin it works !
But recalling the same  plugin in a previous session saved doesnt work.


[17:39:58.491|worker|14204adee80|client:7fc7735e8dd0] adding plugin VST3-HoRNetVUMeterMK4-a607f55...
=> failed with error :  agprocessor|7ff650efb8b8] failed to find plugin descriptor  (previous project I reloaded)

[17:39:49.571|worker|1dcb7c7ee80|client:7fc7d47e4bd0] adding plugin VST3-HoRNetVUMeterMK4-c52cbf37...
=> works ! (new AG insert in the same project on the same remote server)

I dont know what is that added string ( c52cbf37 ) and why does it change when loading the same plugin

**Reply by apohl79** [reply]
the problem is, that the ID of the plugin changed

VST3-HoRNetVUMeterMK4-**a607f55** vs
VST3-HoRNetVUMeterMK4-**c52cbf37**

**Reply by apohl79** [reply]
and if thats the case because of the JUCE upgrade, than your sessions will NOT be backwards compatible with any prior AG version, and thats an issue

**Comment by Henri75** [comment]
I didnt change anything on the server. All the same plugin version for a
while on my local DAW and remote servers.
Clearly it is a Juice ID issue :( Never happened before with any AG version.
It broke compatibility with old sessions ...

Le mer. 13 oct. 2021 à 17:49, Andreas Pohl ***@***.***> a
écrit :

**Reply by apohl79** [reply]
If you can send me your plugin cache files, that would be very helpful, as it does not happen with any of my plugins. Right now this is only an assumption.

**Comment by Henri75** [comment]
Hello,

I am sorry I have already reseted all the cache on all computers.

I will try later to reinstall a previous beta on a computer and generate
the old cache, but currently with more than 1000 plugins installed it may
take even more than an hour to generate each cache...

By the way I tried a new project with beta 6 ... when it works it works
better :) Except for this issue (which needs to rebuild the plugin cache)
and most importantly some crashes.
Today I reopened yesterday project and it crashes cubases ! Seems that a
server is crashing when reloading a plugin, making the AG tray or DAW
plugin plugin crashing too and in cascade crashing cubase. I resetted all
computers and will try to generate some logs and trace.

Last, I still have on some computers a systematic rescan of some plugins (a
few thousands). I checked the log and each of them as a log in success at
the end so I dont know why it rescan everytime.

On an other note, I wanted to congratulate you on AG because it is really
amazing and very promising !
May I ask what is your day job ? and  if you have specific plans for it ?
I have been building/developping startups and raising money for years now,
and I think you could build a business or something with AG if it is what
you wish :)

Best,


Le mer. 13 oct. 2021 à 18:46, Andreas Pohl ***@***.***> a
écrit :

**Comment by Henri75** [comment]
I sent you the hereunder message about the beta 6 crash but it exceeds size
limit on github. So i send it on your other email and open à discussion on
github too

Hello again,
